plugins/modules/role.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, Paul Arthur <paul.arthur@flowerysong.com>
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'XLAB Steampunk'}
+
+DOCUMENTATION = '''
+module: role
+author:
+ - Paul Arthur (@flowerysong)
+ - Manca Bizjak (@mancabizjak)
+ - Aljaz Kosir (@aljazkosir)
+ - Tadej Borovsak (@tadeboro)
+short_description: Manages Sensu roles
+description:
+  - For more information, refer to the Sensu documentation at
+    U(https://docs.sensu.io/sensu-go/latest/reference/rbac/)
+extends_documentation_fragment:
+  - sensu.sensu_go.base
+options:
+  name:
+    description:
+      - The Sensu object's name.
+    type: str
+    required: yes
+  state:
+    description:
+      - Target state of the Sensu object.
+    type: str
+    choices: [ "present", "absent" ]
+    default: present
+  rules:
+    description:
+      - Rules that the role applies.
+    type: list
+    required: yes
+    suboptions:
+      verbs:
+        description:
+          - Permissions to be applied by the rule.
+        type: list
+        required: yes
+      resources:
+        description:
+          - Types of resources the rule has permission to access.
+        type: list
+        required: yes
+      resource_names:
+        description:
+          - Names of specific resources the rule has permission to access.
+        type: list
+        default: []
```

Comment:
Is there a reason why `resouece_names` has a default value set? There is nothing wrong with it. I am just curious where `[]` comes from. But more importantly, the argument spec below does not have this default value set.